_posts/2022-09-06-Umesh-GSoC-on-ConscryptMode.md

```diff
@@ -0,0 +1,24 @@
+---
+title:  "GSoC 2022 final submission for ConscryptMode"
+author: Umesh-01
+---
+
+My name is Umesh Singh and I was an open source contributor through Google Summer of Code this year. Google Summer of Code (GSoC) is a program where external participants can contribute to an open source project over a few months. We learn new computer science concepts, how to work on open source repositories, and create real code contributions to projects!
+
+My project was `Switching Robolectric from BouncyCastle to Conscrypt as the default security provider`. Robolectric was using [BouncyCastle](https://www.bouncycastle.org/) as the Java Cryptography Extension (JCE) security provider. After the introduction of Android P, Android switched to using [Google Conscrypt](https://source.android.com/docs/core/architecture/modular-system/conscrypt) as the security provider. 
+To be more consistent with Android, Robolectric needed to be updated to use Conscrypt as the default security provider.
+ 
+I was already well familiar with the technologies used in the project such as Java, Kotlin, Android Studio, Git and GitHub. I was interested in Android Developement and I was looking for an android project that uses Java as the primary language. I found Robolectric to be a perfect project to contribute to based on my skills, and I was ready to learn about testing concepts since Robolectric is a popular unit testing framework for Android. 
```

Comment:
I would change this paragraph to make it more concise:

`When I encountered Robolectric in the GSOC organization list I know it would be a great fit. I had prior experience with Android development and was looking for growth opportunities in this area. Robolectric also offered a lot of learning opportunities in Java, security, and unit testing concepts.`